core/graphelier-service/models/playback.go

```diff
@@ -0,0 +1,109 @@
+package models
+
+// Modification : Represents an event on a specific order, happening at a given offset time
+type Modification struct {
+	Type     string   `json:"type"`
+	Offset   uint64   `json:"offset"`
+	OrderID  uint64   `json:"order_id"`
+	Price    *float64 `json:"price,omitempty"`
+	Quantity *int64   `json:"quantity,omitempty"`
+	NewID    *uint64  `json:"index,omitempty"`
+	From     *float64 `json:"from,omitempty"`
+	To       *float64 `json:"to,omitempty"`
```

Comment:
why are all these pointers?

Reply:
I wanted to have optional fields in the json response depending on the type of the modification so I opted to use pointers here along with the omitempty marshalling option instead of using different structs